### DISCUSSION TITLE
Sui governor limits

### DISCUSSION THREAD

**Discussion by rmahtani** [wormhole-foundation/wormhole#2890] [Governance] [answered]
I want to open up a discussion for Governor parameters for Sui. We have seen significant outbound flow from Sui and reached the governor limit of $1M/$100K. The current value is limiting outbound transactions from Sui to other chains.

Please consider the current governor limits and if they are appropriate for Sui.

**Comment by brianlong** [comment]
I'd like to see some of the originating transactions on the Sui side. Where can I find those?

**Reply by tbjump** [reply]
You can find transactions that are currently enqueued in the Governor on the Wormhole Dashboard: https://wormhole-foundation.github.io/wormhole-dashboard/

You can follow interactions with the tokenbridge here: https://explorer.sui.io/object/0xc57508ee0d4595e5a8728974a4a93a787d38f339757230d441e895422c07aba9

**Comment by mcf-rocks** [comment]
Do we know what the outflows are about?

If limits need to be raised, then we can do it cautiously maybe, like to $2M initially.

**Reply by nataljakalinin** [reply]
cool

**Comment by claudijd** [comment]
Some data background on real-time Governor status on Sui...

- Flows out of Sui are around 96.79% utilization of the daily limit of $1M for small txs
- There are currently 13 txs in the queue, which include a mix of small txs (held due to small volume increase) and big txs (held due to their size for 24hr hold)
- The big txs have a maximum size of $646K

It looks like Sui's mainnet TB contract addresses have yet to be added to the book, so once that is available, can point to the explorer for the TXs.

**Reply by claudijd** [reply]
**Quick update**: governor saturation level on Sui is remains at virtually 100%

**Reply by claudijd** [reply]
**Quick update**: governor saturation level on Sui is remains at virtually 99.98%

**Reply by claudijd** [reply, marked as answer]
@tbjump proposed a PR to raise the limit from $1M/$100K to $5M/$500K after consultation with Guardians, more details below in this discussion around rationale.

https://github.com/wormhole-foundation/wormhole/pull/2914

**Comment by mcf-rocks** [comment]
If there is nothing untoward going on, we should raise the limits, we are a bridge right - that's our purpose.

**Comment by tbjump** [comment]
I think that the governor limit discussion should be driven by two factors
1. legitimate utilization 
2. security of the network

Everyone should independently verify (1) by looking at individual transactions or checking the overall collateralization status of the bridge. 
Regarding (2), I would highlight that Sui is a pretty new network. I can't find any independent security audits published yet. They do have a bug bounty program, but it was launched only recently. Cross-chain contagion is also further limited by the Global Accountant.

**Comment by brianlong** [comment]
I see several of these TXs are related to "USDC." To my knowledge, Circle has not launched USDC on Sui yet.

Do we have a different symbol for wormhole-wrapped USDC? "wUSDC"? Here is an example: https://explorer.sui.io/txblock/5K3a26wEwZQY7Z5NgBh3iYY1tJMf2UAG55AitYXr69ua

**Reply by brianlong** [reply]
<img width="333" alt="Screenshot 2023-05-08 at 11 20 32 AM" src="https://user-images.githubusercontent.com/79186/236878047-402484f2-a4d1-4e5c-9c07-88d3820f36ea.png">

**Comment by brianlong** [comment]
That same user is trying to move fake BAYCs. 

<img width="298" alt="Screenshot 2023-05-08 at 11 25 04 AM" src="https://user-images.githubusercontent.com/79186/236877725-789c19c3-0196-42a1-b117-dcb0ea09b538.png">

I also noticed someone else bridging out some SUIP tokens (minted earlier today).

**Comment by brianlong** [comment]
I'm inclined to watch this for a while. The governor might be working as intended -- it is possible that some of the teams on Sui will raise a flag.

Do we have any specific community feedback about the larger TXs in the queue?

**Reply by mcf-rocks** [reply]
would be good to sanity check the big ones with a dev from their team... i don't know them at all :/

**Reply by mcf-rocks** [reply]
it is very new, the projects like suiswap are very new.

**Reply by claudijd** [reply]
asked community managers to see if they had any insights

**Comment by Chainode** [comment]
Due to the fact that Sui just recently launched mainnet, we would also be interested to understand better what the outflow actually contains and the validity of it before raising any limits.

**Comment by mcf-rocks** [comment]
it's been 24hrs, nobody screamed yet?

**Comment by HashKeyCloud** [comment]
We believe that safety is the first consideration.

**Comment by d231d** [comment]
It's also worth considering that the notional limit has been reached for several consecutive days. As a result, a significant number of Wormhole users have experienced delays in their transactions being completed, and their voiced concerns suggest a growing frustration. We have received a high volume of user inquiries and support tickets through our official channels on Twitter and Discord. Therefore, to provide users with a better experience using Wormhole and the Portal Bridge, I would advocate for increasing the governor limit for Sui

**Comment by brianlong** [comment]
I checked in with the Sui team; they are unaware of hacks/attacks. I'm comfortable doubling the limit after watching this for the past few days.

**Comment by rmahtani** [comment]
@brianlong is there a particular limit you would advise raising this to?

**Reply by brianlong** [reply]
🤷🏻‍♂️ Do we have written policies for this? Without any guidelines, double the current limits and keep watching for a while.

**Reply by tonyjin** [reply]
No written policy that I'm aware of, but a similar chain (Aptos) has Governor limits of $5M/$500K: https://github.com/wormhole-foundation/wormhole/blob/main/node/pkg/governor/mainnet_chains.go#L33

The governor section of the Wormhole Dashboard (https://wormhole-foundation.github.io/wormhole-dashboard/) show that doubling the current limits to $2M/$200K is likely to be insufficient as there is ~$4.2M total in transactions currently queued to be released tomorrow with 7 queued transactions exceeding >$200K.

With a limit of $5M/$500K, there would still be 2 transactions >$500K queued, but users would have the ability to split their transaction under those limits.

**Reply by brianlong** [reply]
When looking in the Governor section of the Wormhole Dashboard (https://wormhole-foundation.github.io/wormhole-dashboard/), I see a  TX hash for Arbitrum that links out to arbiscan.io.  The TX hashes shown for Sui are not linked, and I can't find the hashes in the Sui explorer. Is that a UI bug in the Wormhole Dashboard?

**Reply by claudijd** [reply]
Adding up the notionals from the dashboard may be misleading on the outbound.  The governor limit as defined is really just "small txs", so if we're trying to find in the queue estimate, we need to discount the "large txs" (anything over the big tx limit for sui, which is currently $100K).  I think the current aggregate waiting of to small txs is actually closer to $800K in "small tx" flow.

There's also a significant amount of big tx flow, which is at the current moment, closer to $10M+ in outbound flows that are subject to the mandatory hold of 24hrs.  The median of these "big txs" are ~$326K and the mean is ~$457k.

So if you were to simply double the limits from $1M/$100K => $2M/$200K you'd be taking in the additional big txs that are in the 100-200k range, which would make the current "small tx" flow to be the current limit ($1M + $800K currently waiting + additional $914K of new small flow = ~$3M in small flow), which means that given current volume, doubling the flow for both small and big txs will not cover the current demand.

Now, if the limits were raised from $1M/$100K => $5M/$500K the math would look like this...

$1M existing flow + $800K waiting flow + additional $4.1M of new small flow == $5.9M in total small flow.  This would cause remaining "large flow", of roughly 7 txs and an additional $6.6M of large flow to be subject to the 24hr big tx hold.  It would still leave you possibly $900K short of existing volume, but it would impact a smaller percentage of users and the wait times on small txs would be faster.

Given those considerations, the $5M level feels more appropriate than the $2M level IMO.  I think what's still outstanding, assuming Guardians agree with the above math is is there enough confidence in Guardian analysis of the existing flows.

**Reply by claudijd** [reply]
> When looking in the Governor section of the Wormhole Dashboard (https://wormhole-foundation.github.io/wormhole-dashboard/), I see a TX hash for Arbitrum that links out to arbiscan.io. The TX hashes shown for Sui are not linked, and I can't find the hashes in the Sui explorer. Is that a UI bug in the Wormhole Dashboard?

I think it just hasn't been added yet.

**Reply by tonyjin** [reply]
I spoke with a contributor familiar with the dashboard and he mentioned there's a bug where the dashboard converts Sui's base58 transaction digest to a 32 byte hex string (Ethereum's format). They should be able to fix this tomorrow and link to the Sui explorer for transactions.

**Reply by tonyjin** [reply]
The contributor fixed the dashboard, you should be able to see correct Sui transaction hashes and links to the explorer now in the Governor section:

![screenshot](https://i.imgur.com/D1sURhW.png)

**Comment by kev1n-peters** [comment]
Here are some of the top tokens transferred from Sui since 5/3/2023 in terms of notional value  (USD). These are rough estimates using today's (5/10/2023) token prices from CoinGecko.
|chain|address|symbol|name|notional (USD)|
|-|-|-|-|-|
|2| 0xdac17f958d2ee523a2206206994597c13d831ec7| USDT| Tether USD|$7,171,357|
|2| 0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48| USDC| USD Coin|$5,653,142|
|2| 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2| WETH| Wrapped Ether|$2,146,235|
|4| 0x8ac76a51cc950d9822d68b83fe1ad97b32cd580d| USDC| USD Coin|$625,583|
|23| 0xfd086bc7cd5c481dcc9c85ebe478a1c0b69fcbb9| USDT| Tether USD|$193,444|
|4| 0x55d398326f99059ff775485246999027b3197955| USDT| Tether USD|$187,646|
|23| 0xff970a61a04b1ca14834a43f5de4533ebddb5cc8| USDC| USD Coin (Arb1)|$145,690|
|5| 0x2791bca1f2de4661ed88a30c99a7a9449aa84174| USDC| USD Coin (PoS)|$138,105|
|1| EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v| USDC| USD Coin|$95,775|
|6| 0xb97ef9ef8734c71904d8002f8b6bc66dd9c48a6e| USDC| USD Coin|$72,027|
|4| 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c| WBNB| Wrapped BNB|$15,892|
|1| Es9vMFrzaCERmJfrF4H2FYD4KCoNkY11McCe8BenwNYB| USDT| USDT|$10,654|
|23| 0x82af49447d8a07e3bd95bd0d56f35241523fbab1| WETH| Wrapped Ether|$9,839|
|24| 0x7f5c764cbc14f9669b88837ca1490cca17c31607| USDC| USD Coin|$9,803|
|5| 0xc2132d05d31c914a87c6611c10748aeb04b58e8f| USDT| (PoS) Tether USD|$9,508|
|1| So11111111111111111111111111111111111111112| SOL| Wrapped SOL|$8,044|
|21| 0x2::sui::SUI| SUI| Sui|$7,180|

**Comment by tbjump** [comment]
I have opened a PR to raise the Sui governor parameters to $5M/$500k. 
There have been no voices suggesting that any of the activity is illegitimate after 3 days of review. 
Based on @claudijd 's analysis, this is what the current demand is roughly like. 
https://github.com/wormhole-foundation/wormhole/pull/2914

**Reply by aadam-10** [reply]
closing discussion thread as quorum was reached on [v2.17.1](https://github.com/wormhole-foundation/wormhole/releases/tag/v2.17.1)
